flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/shuffle/KafkaShuffleITCase.java

```diff
@@ -0,0 +1,309 @@
+import static org.apache.flink.streaming.api.TimeCharacteristic.ProcessingTime;
+import static org.apache.flink.test.util.TestUtils.tryExecute;
+
+/**
+ * Simple End to End Test for Kafka.
+ */
+public class KafkaShuffleITCase extends KafkaConsumerTestBase {
+
+	@BeforeClass
+	public static void prepare() throws Exception {
+		KafkaProducerTestBase.prepare();
+		((KafkaTestEnvironmentImpl) kafkaServer).setProducerSemantic(FlinkKafkaProducer.Semantic.AT_LEAST_ONCE);
+	}
+
+	/**
+	 * Producer Parallelism = 1; Kafka Partition # = 1; Consumer Parallelism = 1.
+	 * To test no data is lost or duplicated end-2-end with the default time characteristic: ProcessingTime
+	 */
+	@Test(timeout = 30000L)
+	public void testSimpleProcessingTime() throws Exception {
+		simpleEndToEndTest("test_simple_processing_time", 100000, ProcessingTime);
+	}
+
+	/**
+	 * Producer Parallelism = 1; Kafka Partition # = 1; Consumer Parallelism = 1.
+	 * To test no data is lost or duplicated end-2-end with time characteristic: IngestionTime
+	 */
+	@Test(timeout = 30000L)
+	public void testSimpleIngestionTime() throws Exception {
+		simpleEndToEndTest("test_simple_ingestion_time", 100000, IngestionTime);
+	}
+
+	/**
+	 * Producer Parallelism = 1; Kafka Partition # = 1; Consumer Parallelism = 1.
+	 * To test no data is lost or duplicated end-2-end with time characteristic: EventTime
+	 */
+	@Test(timeout = 30000L)
+	public void testSimpleEventTime() throws Exception {
+		simpleEndToEndTest("test_simple_event_time", 100000, EventTime);
+	}
+
+	/**
+	 * Producer Parallelism = 2; Kafka Partition # = 3; Consumer Parallelism = 3.
+	 * To test data is partitioned to the right partition with time characteristic: ProcessingTime
+	 */
+	@Test(timeout = 30000L)
+	public void testAssignedToPartitionProcessingTime() throws Exception {
+		testAssignedToPartition("test_assigned_to_partition_processing_time", 100000, ProcessingTime);
+	}
+
+	/**
+	 * Producer Parallelism = 2; Kafka Partition # = 3; Consumer Parallelism = 3.
+	 * To test data is partitioned to the right partition with time characteristic: IngestionTime
+	 */
+	@Test(timeout = 30000L)
+	public void testAssignedToPartitionIngestionTime() throws Exception {
+		testAssignedToPartition("test_assigned_to_partition_ingestion_time", 100000, IngestionTime);
+	}
+
+	/**
+	 * Producer Parallelism = 2; Kafka Partition # = 3; Consumer Parallelism = 3.
+	 * To test data is partitioned to the right partition with time characteristic: EventTime
+	 */
+	@Test(timeout = 30000L)
+	public void testAssignedToPartitionEventTime() throws Exception {
+		testAssignedToPartition("test_assigned_to_partition_event_time", 100000, EventTime);
+	}
+
+	/**
+	 * Schema: (key, timestamp, source instance Id).
+	 * Producer Parallelism = 1; Kafka Partition # = 1; Consumer Parallelism = 1
+	 * To test no data is lost or duplicated end-2-end
+	 */
+	private void simpleEndToEndTest(String topic, int elementCount, TimeCharacteristic timeCharacteristic)
```

Comment:
We use end2end in a different context, where we use a complete Flink distribution to execute the test. 

I'd simply call it `testKafkaShuffle` to avoid any misunderstanding.